### PR TITLE
Upgrade to Bazel 5

### PR DESCRIPTION
**Pull request by lolski**
## What is the goal of this PR?

Make the repository use Bazel 5. Bazel 4 and below only work with the deprecated Python 2, which is no longer included in some operating systems such as the latest Mac OS X.

## What are the changes implemented in this PR?

1. Upgrade Bazel to 5.1.1.
2. Upgrade all `@vaticle` dependencies
3. Update BDD steps as a consequence of upgrading to the latest version of `@vaticle_typedb_behaviour`
4. Update test runners according to the latest version of TypeDB Cluster

### CODE REVIEW

File: test/behaviour/concept/type/thingtype/ThingTypeSteps.ts [hunk 1]
```diff
@@ -105,6 +105,14 @@ When("{root_label}\\({type_label}) set abstract: {bool}", async (rootLabel: Root
     }
 });
 
+When("{root_label}\\({type_label}) set abstract: {bool}; throws exception", async (rootLabel: RootLabel, typeLabel: string, isAbstract: boolean) => {
```
Comment (flyingsilverfin) [inline test/behaviour/concept/type/thingtype/ThingTypeSteps.ts:108]: nice!

**Review by flyingsilverfin — APPROVED** [review]

File: test/behaviour/concept/type/thingtype/ThingTypeSteps.ts (hunk 1 above)
Comment (lolski) [inline test/behaviour/concept/type/thingtype/ThingTypeSteps.ts:108]: As a consequence of updating `@vaticle_typedb_behaviour` to the latest version, we need to implement this newly defined step.

File: test/behaviour/typeql/TypeQLSteps.ts [hunk 2]
```diff
@@ -441,9 +441,9 @@ Then("each answer satisfies", async (template: string) => {
     }
 });
 
-Then("each answer does not satisfy", async (template: string) => {
+Then("templated typeql match; throws exception", async (template: string) => {
```
Comment (lolski) [inline test/behaviour/typeql/TypeQLSteps.ts:444]: Update an existing step as a consequence of updating `@vaticle_typedb_behaviour` to the latest version.

File: test/behaviour/cucumber_test.sh [hunk 3]
```diff
@@ -50,8 +50,8 @@ mkdir ./typedb-distribution/"$DIRECTORY"/typedb_test
 if [[ $PRODUCT == "Core" ]]; then
   ./typedb-distribution/"$DIRECTORY"/typedb server --storage.data typedb_test &
 else
-  ./typedb-distribution/"$DIRECTORY"/typedb cluster --address "127.0.0.1:1729:1730:1731" --encryption-enabled=true &
-  ROOT_CA=`realpath ./typedb-distribution/"$DIRECTORY"/server/conf/encryption/rpc-root-ca.pem`
+  ./typedb-distribution/"$DIRECTORY"/typedb cluster --server.encryption.enable=true &
```
Comment (lolski) [inline test/behaviour/cucumber_test.sh:53]: Update TypeDB Cluster bootup command according to the latest syntax.

File: test/common/start-cluster-servers.sh [hunk 4]
```diff
@@ -22,17 +22,35 @@
 
 set -e
 
+function server_start() {
```
Comment (lolski) [inline test/common/start-cluster-servers.sh:25]: Update TypeDB Cluster bootup command according to the latest syntax.

File: test/integration/test-cluster-failover.js [hunk 5]
```diff
@@ -44,6 +44,25 @@ function getServerPID(port) {
     else throw new Error("Found multiple PIDs: '" + serverPID + " for port: '" + port + "'");
 }
 
+function serverStart(idx) {
```
Comment (lolski) [inline test/integration/test-cluster-failover.js:47]: Update TypeDB Cluster bootup command according to the latest syntax.

**Pull request merged**